pkg/loki/loki.go

```diff
@@ -694,13 +694,12 @@ func (t *Loki) setupModuleManager() error {
 	}
 
 	// Add bloom gateway ring in client mode to IndexGateway service dependencies if bloom filtering is enabled.
-	if t.Cfg.isModuleEnabled(IndexGateway) && t.Cfg.BloomGateway.Enabled {
```

Comment:
I think checking for `t.Cfg.isModuleEnabled(IndexGateway)` is incorrect here. If we run the `read` or `backend` target it won't work (i.e. only works in microservice mode).

Additionally, we'll always need to add the bloom gateway ring dependency to the index gateway regardless of the target we are running.